op-program: Build an interop prestate

ajsutton opened the pull request:
**Description**

The bootstrap process for interop is different so introduces a separate main method to set the interop mode flag.

Previously had planned to use an env var but setting it in the cannon prestate wound up being too deep in the weeds of MIPS so going with a simpler approach.

Builds using cannon-mt64 since we're moving that way and planning to upgrade to it prior to interop anyway.

ajsutton committed:
op-program: Build an interop prestate

The bootstrap process for interop is different so introduces a separate main method.

Inphi reviewed (commented):
I do prefer having another entry point. It separates interop concerns from the VM.

However, we still need the `OP_PROGRAM_CLIENT_USE_INTEROP` envar. So that block execution hint can invoke the "interop-build" op-program to run pre-interop derivation.

We can have both. If `OP_PROGRMA_CLIENT_USE_INTEROP=false`, then always run regular derivation.

ajsutton committed:
op-program: Stop setting USE_INTEROP env var when running client in external process.

The interop version is used by specifying the interop client binary.
Update Makefile to build a native version of the interop client binary.

ajsutton commented:
> However, we still need the OP_PROGRAM_CLIENT_USE_INTEROP envar. So that block execution hint can invoke the "interop-build" op-program to run pre-interop derivation.

Can you clarify where this is used? I can't see anywhere that reads the env var anymore.

I did find we were still setting it when op-program ran the client as an external executable but that's not needed as you should just specify the interop executable. So have removed setting the env var there and now there are no references to it at all that I can see.

Inphi commented:
> > However, we still need the OP_PROGRAM_CLIENT_USE_INTEROP envar. So that block execution hint can invoke the "interop-build" op-program to run pre-interop derivation.
> 
> Can you clarify where this is used? I can't see anywhere that reads the env var anymore.
> 
> I did find we were still setting it when op-program ran the client as an external executable but that's not needed as you should just specify the interop executable. So have removed setting the env var there and now there are no references to it at all that I can see.

Actually I think this is what we want. I was thinking we'd wanna run a pre-interop program for the block-exec hint. But that could never work because the host pre-configured prestate outputs are incompatible with pre-interop execution.

We could also extract the pre-interop agreed output from the interop prestate and use that to configure a pre-interop execution. Honestly, this is probably the safest to avoid the possibility of recursive program execution. I'll look into this more carefully in another PR.
